sunpy/image/transform.py

```diff
@@ -126,8 +130,27 @@ def affine_transform(image, rmatrix, order=3, scale=1.0, image_center=None,
                           SunpyUserWarning)
             adjusted_image = np.nan_to_num(adjusted_image)
 
+        # Scale image to range [0, 1] if it is valid (not made up entirely of NaNs)
+        is_nan_image = np.all(np.isnan(adjusted_image))
+        if is_nan_image:
+            adjusted_missing = missing
+        else:
+            im_min = np.nanmin(adjusted_image)
+            adjusted_image -= im_min
+            im_max = np.nanmax(adjusted_image)
+            if im_max > 0:
```

Comment:
Why this if clause? If the minimum value in the image is now zero (by definition because you have subtracted it), surely the maximum value will always be greater than zero?

Reply:
From what I can gather, that code was introduced [here](https://github.com/sunpy/sunpy/commit/982bb0dadd736a6c7d91defb596f4dc4cdcd1679). If we have a flat array with the same values, after its minimum is subtracted, we're left with an array of zeros. It's maximum is evaluted to zero and a 0/0 division occurs in the next line.
This code handles that 😃

And yes, I'll fix the formatting issues, thanks! 👍